server/v2/cometbft/handlers/handlers.go

```diff
@@ -13,16 +13,16 @@ import (
 type (
 	// PrepareHandler passes in the list of Txs that are being proposed. The app can then do stateful operations
 	// over the list of proposed transactions. It can return a modified list of txs to include in the proposal.
-	PrepareHandler[T transaction.Tx] func(context.Context, AppManager[T], transaction.Codec[T], *abci.PrepareProposalRequest) ([]T, error)
+	PrepareHandler[T transaction.Tx] func(context.Context, AppManager[T], transaction.Codec[T], *abci.PrepareProposalRequest, string) ([]T, error)
```

Comment:
before the chainID addition, having unnamed parameters made sense as the type was self explanatory. Now given the `chainID` is `string` I think it is best to name all those arguments, so you can actually see what it means.